stdlib/private/TensorFlowUnittest/TensorFlowUnittestRemote.swift

```diff
@@ -0,0 +1,109 @@
+#if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
+import Darwin
+#else
+import Glibc
+#endif
+import CTensorFlow
+import CTensorFlowUnittest
+import TensorFlow
+import StdlibUnittest
+
+typealias CTFServer = OpaquePointer
+
+@usableFromInline
+func debugLog(_ message: @autoclosure () -> String,
+              file: StaticString = #file,
+              line: UInt = #line) {
+  if _RuntimeConfig.printsDebugLog {
+    print("[\(file):\(line)] \(message())")
+    // This helps dump more log before a crash.
+    fflush(stdout)
+  }
+}
+
+
+private class TestCluster {
+  var num_tasks:Int = 0
+  var servers: [CTFServer] = []
+  public var serverDef:String = ""
+
+  public init(num_tasks: Int) {
+    self.num_tasks = num_tasks
+    self.serverDef = startServers()
+  }
+
+  deinit {
+    // TODO: There is no clean way to shutdown GrpcServers yet.
+  }
+
+  private func getServerDef(tasks: String, task_index: Int) -> String {
+    return  """
+      cluster {
+        job {
+          name: "localhost"
+          \(tasks)
+        }
+      }
+      job_name: "localhost"
+      task_index: \(task_index)
+      protocol: "grpc"
+      """
+  }
+
+  private func getTasks() -> String {
+    var tasks = ""
+    for i in 0..<num_tasks {
+      let port = TF_PickUnusedPortOrDie()
+      let task = """
+          tasks {
+            key: \(i)
+            value: "localhost:\(port)"
+          }
+      """
+      debugLog("Picking port \(port) for \(i)...");
+      tasks += task
+    }
+    return tasks;
+  }
+
+  private func startServers() -> String {
+    let tasks = getTasks()
+    let status = TF_NewStatus()
+    // Start everything except the 0-th task, which will
+    // be started when we set TFE_ContextSetServerDef.
+    for i in 1..<num_tasks {
+      debugLog("Starting task \(i)...");
+      let serverDefText = getServerDef(tasks: tasks, task_index: i)
+      let serverDef = TFE_GetServerDef(serverDefText, status)
+      checkOk(status)
+      let server = TF_NewServer(
```

Comment:
the code could look a bit nicer if we use implicitly unwrapped optional  (https://krakendev.io/blog/when-to-use-implicitly-unwrapped-optionals). I believe the code would be:

```swift
let server: CTFServer! = TF_NewServer(...)
checkOk(status)
// when we access `server`, it'll be implicitly unwrapped (i.e., `server!`)
TF_ServerStart(server, status)
...
```

Reply:
same for `serverDef` in the next file.

Reply:
+1

Reply:
To use implicit unwrapping, shouldn't the value guaranteed to be non-nil? TF_NewServer may return nullptr in some cases.

Reply:
I believe as long as the returned value of IUO is not being read, it can stay nil. This is the case for us because we always check status first.

Reply:
Done.